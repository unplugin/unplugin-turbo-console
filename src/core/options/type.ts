export interface Options {
  /**
   * Add a string prefix to the console output.
   *
   * @see https://utc.yuy1n.io/features/custom-prefix.html
   * @default undefined
   */
  prefix?: string
  /**
   * Add a string suffix to the console output.
   *
   * @see https://utc.yuy1n.io/features/custom-prefix.html
   * @default undefined
   */
  suffix?: string
  /**
   * Launch editor option
   *
   * @see https://utc.yuy1n.io/features/launch-editor.html
   * @default true
   */
  launchEditor?: boolean | LaunchEditorOption
  /**
   * Highlight option
   *
   * @see https://utc.yuy1n.io/features/highlight.html
   * @default true
   */
  highlight?: boolean | HighlightOption
  /**
   * Enable pass logs feature.
   *
   * @see https://utc.yuy1n.io/features/pass-logs.html
   * @default true
   */
  passLogs?: boolean
  /**
   * Enable inspector feature.
   *
   * @see https://utc.yuy1n.io/features/inspector.html
   * @default
   * {
   *   printUrl: true
   * }
   */
  inspector?: boolean | InspectorOption
  /**
   * Custom server option.
   *
   * @default { port: 3070, host: '127.0.0.1' }
   */
  server?: {
    port?: number
    host?: string
  }
}

interface LaunchEditorOption {
  /**
   * The specific editor to launch.
   *
   * @see https://github.com/yyx990803/launch-editor#supported-editors
   * @default undefined
   *
   */
  specifiedEditor?: string
}

interface HighlightOption {
  /**
   * Whether to show extended path name when the file's (or folder's) name contains an element in the array.
   *
   * @see https://utc.yuy1n.io/features/highlight.html#expand-path-file-name
   * @default undefined
   *
   */
  extendedPathFileNames?: string[]
  /**
   * Whether to detect the system theme and adjust the highlight color.
   *
   * @see https://utc.yuy1n.io/features/highlight.html#theme-detect
   * @default false
   *
   */
  themeDetect?: boolean
}

interface InspectorOption {
  /**
   * Print inspector url in the console when server start.
   *
   * @default true
   */
  printUrl?: boolean
}
