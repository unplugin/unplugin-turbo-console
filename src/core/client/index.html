<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Launch Editor</title>
</head>
<style>
  #error-info {
    color: red;
    font-size: 2rem;
  }

  body {
    background-color: white;
  }

  @media (prefers-color-scheme: dark) {
    body {
      background-color: black;
    }
  }
</style>
<body>
  <div id="error-info"></div>
</body>
<script>
 (async () => {
   try {
     const hash = window.location.hash
     const decodeBase64 = window.atob(hash.slice(1))
     await fetch(`/__open-in-editor?file=${encodeURIComponent(decodeBase64)}`)
     window.close()
   }
   catch (error) {
     const errorInfo = document.getElementById('error-info')
     errorInfo.innerText = String(error)
   }
 })()
</script>
</html>
