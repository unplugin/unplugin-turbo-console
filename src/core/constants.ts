export const PLUGIN_NAME = 'unplugin-turbo-console'
export const NUXT_CONFIG_KEY = 'turboConsole'

export const VirtualModules = {
  Init: '~console',
  ThemeDetect: '~console/theme-detect',
  VueDevTools: '~console/vue-devtools',
  ServerInfo: '~console/server-info',
}

export const PLUGIN_SERVER_DEFAULT_PORT = 3070
export const PLUGIN_SERVER_PORT_RANGE: [number, number] = [PLUGIN_SERVER_DEFAULT_PORT, 6000]
