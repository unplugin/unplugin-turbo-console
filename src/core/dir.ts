import { fileURLToPath } from 'node:url'
import { dirname, resolve } from 'pathe'

export const DIR_DIST = typeof __dirname !== 'undefined'
  ? __dirname
  : dirname(fileURLToPath(import.meta.url))

export const CLIENT_DIR = resolve(DIR_DIST, './client')
