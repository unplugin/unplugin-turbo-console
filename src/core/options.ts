import { env } from 'node:process'
import type { Options } from '../types'

const DETAULT_OPTIONS: Options = {
  prefix: '',
  suffix: '',
  disableLaunchEditor: false,
  disableHighlight: false,
  extendedPathFileNames: [],
  port: 3070,
}

export const BUILD_OPTIONS: Options = {
  disableLaunchEditor: true,
  disableHighlight: true,
}

export function resolveOptions(options: Options): Options {
  let resolved = {
    ...DETAULT_OPTIONS,
    ...options,
  }

  if (env.NODE_ENV === 'production') {
    resolved = {
      ...resolved,
      ...BUILD_OPTIONS,
    }
  }

  return resolved
}
