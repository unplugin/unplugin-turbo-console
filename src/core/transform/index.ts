import type { Context } from './../../types'
import MagicString from 'magic-string'
import { parseSync } from 'oxc-parser'
import { calculateStart, genConsoleString, getCompiler, isConsoleExpression, isPluginDisable } from '../utils'
import { walk } from '../utils/walker'
import { compilers } from './compilers'

export async function transform(context: Context) {
  const { code, id, options } = context
  const magicString = new MagicString(code)

  const compiler = await getCompiler(id)

  if (!compiler) {
    return {
      code: magicString.toString(),
      map: magicString.generateMap({
        source: id,
        file: id,
        includeContent: true,
        hires: true,
      }),
    }
  }

  const compileResult = await compilers[compiler](context)

  const oxcParsedResult = parseSync(id, compileResult.script, {
    lang: (compileResult.lang || 'js') as 'js' | 'jsx' | 'ts' | 'tsx' | undefined,
    sourceType: 'module',
  })

  const { program, comments = [], magicString: oxcMs } = oxcParsedResult

  if (isPluginDisable({
    comments,
    originalLine: 1,
    id,
    type: 'top-file',
    compiler,
    oxcMs,
  })) {
    return {
      code: magicString.toString(),
      map: magicString.generateMap({
        source: id,
        file: id,
        includeContent: true,
        hires: true,
      }),
    }
  }

  await walk(program, {
    enter(node) {
      if (isConsoleExpression(node)) {
        const originalExpression = oxcMs.getSourceText(node.start, node.end)

        if (originalExpression.includes('%c'))
          return false

        const { line, column } = oxcMs.getLineColumnNumber(node.start)

        const originalLine = line + compileResult.line + 1
        const originalColumn = column

        if (isPluginDisable({
          comments,
          originalLine: line,
          id,
          type: 'inline-file',
          compiler,
          oxcMs,
        })) {
          return false
        }

        const consoleMethod = (node as any).callee.property.name
        const args = (node as any).arguments

        const argsStart = calculateStart(compileResult.script, oxcMs.getLineColumnNumber(args[0].start))
        const argsEnd = calculateStart(compileResult.script, oxcMs.getLineColumnNumber(args[args.length - 1].end))

        const argType = args[0].type

        const argsName = oxcMs.getSourceText(args[0].start, args[args.length - 1].end)
          .toString()
          .replace(/`/g, '')
          .replace(/\n/g, '')
          .replace(/"/g, '')

        const { consoleString, _suffix } = genConsoleString({
          options,
          consoleMethod,
          originalLine,
          originalColumn,
          argType,
          argsName,
          id,
        })

        consoleString && magicString.appendLeft(argsStart + compileResult.offset, consoleString)
        _suffix && magicString.appendRight(argsEnd + compileResult.offset, `,"${_suffix}"`)
      }
    },
  })

  return {
    code: magicString.toString(),
    map: magicString.generateMap({
      source: id,
      file: id,
      includeContent: true,
      hires: true,
    }),
  }
}
