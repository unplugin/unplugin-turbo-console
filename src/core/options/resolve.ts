import type { Options } from './type'
import { env } from 'node:process'
import { PLUGIN_SERVER_DEFAULT_PORT } from '../constants'

export function resolveOptions(options: Partial<Options> = {}): Required<Options> {
  const resolved = {
    launchEditor: options.launchEditor ?? true,
    highlight: options.highlight ?? true,
    passLogs: options.passLogs ?? true,
    inspector: options.inspector ?? { printUrl: true },
    server: {
      port: options.server?.port ?? PLUGIN_SERVER_DEFAULT_PORT,
      host: options.server?.host ?? '127.0.0.1',
    },
    prefix: options.prefix ?? '',
    suffix: options.suffix ?? '',
  }

  if (env.NODE_ENV === 'production') {
    resolved.launchEditor = false
    resolved.highlight = false
    resolved.passLogs = false
    resolved.inspector = false
  }

  return resolved
}
