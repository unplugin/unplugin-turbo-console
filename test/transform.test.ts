import { join } from 'pathe'
import { describe, expect, it, vi } from 'vitest'
import { resolveOptions } from '../src/core/options'
import { transform } from '../src/core/transform'
import { INCLUDES_HIGHLIGHT, SVELTE, TSX, TYPESCRIPT, UTF_8, VUE_OPTIONS, VUE_SCRIPT_SETUP, VUE_SETUP_NO_LANG, WIN_PATH } from './fixtures/transform'

vi.mock('node:process', () => {
  return {
    cwd: vi.fn(() => join('/', 'mock', 'path')),
    env: {
      NODE_ENV: 'development',
    },
  }
})

const mockFilePathMap = new Map()
mockFilePathMap.set('../../home/runner/App.vue', 'appvue')
mockFilePathMap.set('../../home/runner/main.ts', 'maints')
mockFilePathMap.set('../../home/runner/main.js', 'mainjs')
mockFilePathMap.set('../../home/runner/+page.svelte', 'pagtsx')
mockFilePathMap.set('../../home/runner/page.tsx', 'svelte')

globalThis.TurboConsoleFilePathMap = mockFilePathMap

describe('vue transform', () => {
  it ('script setup', async () => {
    VUE_SCRIPT_SETUP.options = resolveOptions(VUE_SCRIPT_SETUP.options)
    expect(await transform(VUE_SCRIPT_SETUP)).matchSnapshot()
  })

  it('options', async () => {
    VUE_OPTIONS.options = resolveOptions(VUE_OPTIONS.options)
    expect(await transform(VUE_OPTIONS)).matchSnapshot()
  })

  it('script setup no lang', async () => {
    VUE_SETUP_NO_LANG.options = resolveOptions(VUE_SETUP_NO_LANG.options)
    expect(await transform(VUE_SETUP_NO_LANG)).matchSnapshot()
  })
})

describe('vanilla transform', () => {
  it('typescript', async () => {
    TYPESCRIPT.options = resolveOptions(TYPESCRIPT.options)
    expect(await transform(TYPESCRIPT)).matchSnapshot()
  })

  it('tsx', async () => {
    TSX.options = resolveOptions(TSX.options)
    expect(await transform(TSX)).matchSnapshot()
  })
})

describe('svelte transform', () => {
  it('svelte', async () => {
    SVELTE.options = resolveOptions(SVELTE.options)
    expect(await transform(SVELTE)).matchSnapshot()
  })
})

describe('edge case', () => {
  it('includes highlight', async () => {
    INCLUDES_HIGHLIGHT.options = resolveOptions(INCLUDES_HIGHLIGHT.options)
    expect(await transform(INCLUDES_HIGHLIGHT)).matchSnapshot()
  })

  it('win path', async () => {
    WIN_PATH.options = resolveOptions(WIN_PATH.options)
    expect(await transform(WIN_PATH)).matchSnapshot()
  })
})

describe('utf-8', () => {
  it('includes utf-8', async () => {
    UTF_8.options = resolveOptions(UTF_8.options)
    expect(await transform(UTF_8)).toMatchSnapshot()
  })
})
