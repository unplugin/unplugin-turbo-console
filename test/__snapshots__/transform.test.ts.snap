// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`edge case > includes highlight 1`] = `
{
  "code": "console.log('%cLog Message', 'color: orange');",
  "map": SourceMap {
    "file": "main.js",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC",
    "names": [],
    "sources": [
      "main.js",
    ],
    "sourcesContent": [
      "console.log('%cLog Message', 'color: orange');",
    ],
    "version": 3,
  },
}
`;

exports[`edge case > win path 1`] = `
{
  "code": "<template>
  Hello World
</template>

<script setup lang="ts">
  const msg: string = "Hello World"
  console.log("%c🚀 App.vue·7 ~ msg%c🔦 http://127.1:3070#appvue,7,3",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#fff;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#4FC08D90;' : '#4FC08D;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n",msg)
</script>
",
  "map": SourceMap {
    "file": "App.vue",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACT,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACZ,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;;AAEV,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACvB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAClC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,gUAAC,CAAC,CAAC,CAAC;AACjB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;",
    "names": [],
    "sources": [
      "App.vue",
    ],
    "sourcesContent": [
      "<template>
  Hello World
</template>

<script setup lang="ts">
  const msg: string = "Hello World"
  console.log(msg)
</script>
",
    ],
    "version": 3,
  },
}
`;

exports[`svelte transform > svelte 1`] = `
{
  "code": "<script lang="ts">
  // xxxx
  const a: string = 'Hello'
  console.log("%c🚀 +page.svelte·4 ~ a%c🔦 http://127.1:3070#pagtsx,4,3",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#fff;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#FF3E0090;' : '#FF3E00;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n",a)
</script>

<h1>Welcome to SvelteKit</h1>
<p>
  Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation
</p>",
  "map": SourceMap {
    "file": "+page.svelte",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACjB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACR,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC1B,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,mUAAC,CAAC;AACf,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;;AAER,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC5B,CAAC,CAAC;AACF,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACpF,CAAC,CAAC,CAAC",
    "names": [],
    "sources": [
      "+page.svelte",
    ],
    "sourcesContent": [
      "<script lang="ts">
  // xxxx
  const a: string = 'Hello'
  console.log(a)
</script>

<h1>Welcome to SvelteKit</h1>
<p>
  Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation
</p>",
    ],
    "version": 3,
  },
}
`;

exports[`utf-8 > includes utf-8 1`] = `
{
  "code": "const a = '你好' 
console.log("%c🚀 main.js·2 ~ a%c🔦 http://127.1:3070#mainjs,2,1",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#111827;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#F7DF1E90;' : '#F7DF1E;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n",a)",
  "map": SourceMap {
    "file": "main.js",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACd,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,iUAAC,CAAC",
    "names": [],
    "sources": [
      "main.js",
    ],
    "sourcesContent": [
      "const a = '你好' 
console.log(a)",
    ],
    "version": 3,
  },
}
`;

exports[`vanilla transform > tsx 1`] = `
{
  "code": ""use client";

  export default function Home() {
    console.log("%c🚀 page.tsx·4%c🔦 http://127.1:3070#svelte,4,5",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#fff;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#3178C690;' : '#3178C6;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n","hello nextjs");
    function hello() {
      console.log("%c🚀 page.tsx·6%c🔦 http://127.1:3070#svelte,6,7",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#fff;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#3178C690;' : '#3178C6;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n","hello nextjs");
    }
  
    return <main>page.tsx</main>;
  }
  ",
  "map": SourceMap {
    "file": "page.tsx",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;;AAEZ,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACjC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,2TAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC/B,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACrB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,2TAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACjC,CAAC,CAAC,CAAC,CAAC;AACJ,CAAC;AACD,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAChC,CAAC,CAAC;AACF,CAAC",
    "names": [],
    "sources": [
      "page.tsx",
    ],
    "sourcesContent": [
      ""use client";

  export default function Home() {
    console.log("hello nextjs");
    function hello() {
      console.log("hello nextjs");
    }
  
    return <main>page.tsx</main>;
  }
  ",
    ],
    "version": 3,
  },
}
`;

exports[`vanilla transform > typescript 1`] = `
{
  "code": "const msg: string = "hello TypeScript"
  console.log("%c🚀 main.ts·2 ~ msg%c🔦 http://127.1:3070#maints,2,3",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#fff;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#3178C690;' : '#3178C6;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n",msg)
  ",
  "map": SourceMap {
    "file": "main.ts",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACrC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,gUAAC,CAAC,CAAC,CAAC;AACjB,CAAC",
    "names": [],
    "sources": [
      "main.ts",
    ],
    "sourcesContent": [
      "const msg: string = "hello TypeScript"
  console.log(msg)
  ",
    ],
    "version": 3,
  },
}
`;

exports[`vue transform > options 1`] = `
{
  "code": "<script>
  export default {
    data() {
      return {
        greeting: 'Hello World!'
      }
    },
    mounted() {
      const msg = "hello vue"
      console.log("%c🚀 App.vue·10 ~ msg%c🔦 http://127.1:3070#appvue,10,7",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#fff;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#4FC08D90;' : '#4FC08D;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n",msg)
    }
  }
  </script>
  
  <template>
    <p class="greeting">{{ greeting }}</p>
  </template>
",
  "map": SourceMap {
    "file": "App.vue",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACP,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACjB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACX,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACb,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC/B,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACN,CAAC,CAAC,CAAC,CAAC,CAAC;AACL,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACd,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC5B,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,kUAAC,CAAC,CAAC,CAAC;AACrB,CAAC,CAAC,CAAC,CAAC;AACJ,CAAC,CAAC;AACF,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACV,CAAC;AACD,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACX,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACzC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;",
    "names": [],
    "sources": [
      "App.vue",
    ],
    "sourcesContent": [
      "<script>
  export default {
    data() {
      return {
        greeting: 'Hello World!'
      }
    },
    mounted() {
      const msg = "hello vue"
      console.log(msg)
    }
  }
  </script>
  
  <template>
    <p class="greeting">{{ greeting }}</p>
  </template>
",
    ],
    "version": 3,
  },
}
`;

exports[`vue transform > script setup 1`] = `
{
  "code": "<template>
  Hello World
</template>

<script setup lang="ts">
  const msg: string = "Hello World"
  console.log("%c🚀 App.vue·7 ~ msg%c🔦 http://127.1:3070#appvue,7,3",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#fff;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#4FC08D90;' : '#4FC08D;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n",msg)
</script>
",
  "map": SourceMap {
    "file": "App.vue",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACT,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACZ,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;;AAEV,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACvB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAClC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,gUAAC,CAAC,CAAC,CAAC;AACjB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;",
    "names": [],
    "sources": [
      "App.vue",
    ],
    "sourcesContent": [
      "<template>
  Hello World
</template>

<script setup lang="ts">
  const msg: string = "Hello World"
  console.log(msg)
</script>
",
    ],
    "version": 3,
  },
}
`;

exports[`vue transform > script setup no lang 1`] = `
{
  "code": "<template>
  Hello World
</template>

<script setup>
  const msg = "Hello World"
  console.log("%c🚀 App.vue·7 ~ msg%c🔦 http://127.1:3070#appvue,7,3",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#fff;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#4FC08D90;' : '#4FC08D;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n",msg)
</script>
",
  "map": SourceMap {
    "file": "App.vue",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACT,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACZ,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;;AAEV,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACb,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC1B,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,gUAAC,CAAC,CAAC,CAAC;AACjB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;",
    "names": [],
    "sources": [
      "App.vue",
    ],
    "sourcesContent": [
      "<template>
  Hello World
</template>

<script setup>
  const msg = "Hello World"
  console.log(msg)
</script>
",
    ],
    "version": 3,
  },
}
`;
